### DISCUSSION TITLE
Ideas about better save options for custom pirate (enemy) submarines

### DISCUSSION THREAD

**Discussion by Singularia** [FakeFishGames/Barotrauma#5945] [Ideas and suggestions]
Inspired by personal experience and Regalis' reply on issue https://github.com/Regalis11/Barotrauma/issues/5766

Current system of selecting pirate submarines depending on the difficulty is good in terms that it will eliminate moments when on higher difficulties players will fight some “dugongs and shuttles” instead of more dangerous submarines like Typhon 2.
The downside is that it will require modification of a missions list for each new submarine and not everyone know how (or want) to do it.

I suggest these changes:
![image](https://user-images.githubusercontent.com/62324831/120271942-f3ec6a80-c2f7-11eb-8f71-a1387fb22f49.png)

1. Remove hardcoded list of submarines from Mission.xml and replace it with difficulty tags ("easy", "medium" and "hard" for example) and raw difficulty values (if preferred difficulty <=20 then “easy”, if >=70 then "hard" etc.)
2. Adding ability to set such tags to pirate submarines in the sub editor
These changes will allow sub builders easily make new pirate submarines without need to add “new” missions for each submarine.

Additional features that would be a good addition:

1. **Commonness** of an appearance. Will help set how often submarine will appear compared to others in similar difficulty level (Dugong will appear less compared to Hemulen for example)
2. **Crew size**. Will allow builders set how many pirates will be on the sub
2a. **Crew Experience**. Will allow builders to set how experienced (means difficult to kill and skilled) pirate crew will be in general (can be presets like “Rookie” “Advanced” “Experienced” with different skill levels and gear)
3. Biome types. Will allow builders to set in what biomes you'll be able to encounter particular pirate submarine. (For example, pirate Selkie will appear only in the Cold Caves, Europan Ridge and Aphotic Plateau and Humpback will be appear in all biomes)

Also feature that is not related to pirate subs but to sub editor in general would be an additional button to remove preview image, for now we have ability to create preview and set it from existing image file but we can’t remove it entirely (current workaround is a blank 8x8 PNG image). This will be useful mostly for Beacons, Wrecks, Pirate subs and Drones/Shuttles because they don’t require preview image and this will help decrease size of a .SUB file.

**Comment by PanmanS** [comment]
This looks much better than doing stuff in `missions.xml`!

**Comment by Jade-Harleyy** [comment]
#10288 👀
